requirements-dev.txt

```diff
@@ -1,14 +1,763 @@
-isort~=5.11
-autopep8~=1.6
-flake8~=6.1
-flake8-docstrings~=1.7
-flake8-builtins~=2.1
-pyproject-flake8~=6.1
-pep8-naming~=0.13
-pytest~=7.2
-pyright==1.1.345
-pytest-operator~=0.23
-coverage[toml]~=7.0
-typing_extensions~=4.2
+#
```

Comment:
No, and actually even the `requirements.txt` isn't strictly needed. I did more reading on this, and there seems to be mixed opinions on putting the lock file in source control when (a) it's for a library and (b) it's Python.

If we consider what value it's providing: it's saying what dependencies were used when CI ran the tests, particularly when a build was done. However, we have that information in the runner output as well. If downstream users want to know what dependencies are being used then it's not our lock file that matters, it's theirs. So it's really only for people working on ops, to avoid inconsistent behaviour due to their environments being different. It seems like we would probably actually want to hit that problem so that we know that it should be handled (e.g. avoid some specific version) and can figure it out by just comparing `freeze` output.

So, on balance, I think I'm back to not having these in source control at all (which is what I had in some early version of this PR).